lib/controllers/in-memory-db.controller.ts

```diff
@@ -0,0 +1,71 @@
+import { Get, Post, Put, Delete, Body, Param } from '@nestjs/common';
+import { InMemoryDBService } from '../services';
+import { InMemoryDBEntity } from '../interfaces';
+import { Observable } from 'rxjs';
+
+/**
+ * @example
+ *
+ * ```ts
+ * @Controller('api/user')
+ * class UsersController extends InMemoryDbController<User> {
+ *
+ *   constructor(private dbService: InMemoryDBService<User>) {
+ *     super(dbService);
+ *   }
+ *
+ * }
+ *
+ * ```
+ */
+export abstract class InMemoryDbEntityController<T extends InMemoryDBEntity> {
```

Comment:
```suggestion
export abstract class InMemoryDbEntityAsyncController<T extends InMemoryDBEntity> {
```


create a second one that uses the sync methods and is called `InMemoryDbEntityController`